src/hooks/useNotifications.ts

```diff
@@ -146,28 +141,47 @@ export const useNotifications = (colors: boolean): NotificationsState => {
                               )
                             : accounts.token;
 
-                          const cardinalData = (
-                            await apiRequestAuth(
-                              notification.subject.url,
-                              'GET',
-                              token,
-                            )
-                          ).data;
-
-                          const state =
-                            cardinalData.state === 'closed'
-                              ? cardinalData.state_reason ||
-                                (cardinalData.merged && 'merged') ||
-                                'closed'
-                              : (cardinalData.draft && 'draft') || 'open';
-
-                          return {
-                            ...notification,
-                            subject: {
-                              ...notification.subject,
-                              state,
-                            },
-                          };
+                          switch (notification.subject.type) {
```

Comment:
Moving this code out, as you suggest, will make a big difference to the mind-boggling level of indentation here :)

Reply:
https://github.com/gitify-app/gitify/issues/829 for future us